ieeeuoft/hackathon-template
# RSVP View Dashboard Implementation

## Overview

- Resolves #181 
- Adds functionality to the user dashboard that states the status of their application
- Adds various states to the application processes, seen in the `get_context_data` function in  `event/views.py`. These states cover the entire application processes.
- Adds unit tests that cover the states for a user's application process.
- Fixed an issue where `num_members` on a team wasn't showing a value after registration closed.
- Added `WAITLIST_ACCEPTANCE_START_TIME` which is the time at which waitlisted people will start being accepted into the event. This will show up on the dashboard of waitlisted persons.
- Also Resolves #156, forcing only PDFs to show up on your file explorer when you're uploading a resume, per @lisa-sa-li

## Unit Tests Created

- Wrote unit tests for the following states an application can be in
   - Applications Closed
   - Application Complete
   - Application Incomplete
   - Application reviewed but no decision sent out yet
   - Accepted, awaiting RSVP (the buttons here too are messed up)
   - Waitlisted
   - Rejected
   - Offer Accepted
   - Offer Declined
   - RSVP deadline passed

## Steps to QA

- Go into the website and make a new account `(X)` and activate it. When you go to your dashboard you should see your status as `Application Incomplete`
- Go and do your application. You should now see your status as `Application Complete`
- Go to the admin site and create a review for the new account `(X)`. Set the status to `Rejected`, and `decision_sent_date` to Today; the other fields don't matter. You should now see your status on the dashboard of `(X)` as `Rejected`
- Go to the admin site and edit the review for `(X)`. Set the status to `Waitlisted`. You should now see your status on the dashboard of `(X)` as `Waitlisted`
   - If you're feeling adventurous, play around with the `hours` in the `timedelta` of the `WAITLISTED_ACCEPTANCE_START_TIME` settings variable to make sure the offset is correct.

- Go to the admin site and edit the review for `(X)`. Set the status to `Accepted`. You should now see your status on the dashboard of `(X)` as `Accepted, awaiting RSVP`
- On your dashboard click the `WILL ATTEND (ACCEPT)` button. You should now see your status on the dashboard change to `Offer Accepted`
- Now click the `CANNOT ATTEND (DECLINE)` button. You should now see your status on the dashboard change to `Offer Declined`
- Now go to the admin site and edit the review for `(X)` - change the `decision_sent_date` to `RSVP_DAYS + 1` in the past. Go back to the dashboard for `(X)`. You should now not be able to see the buttons for accepting or declining your RSVP.

- Lisa did a fix for allowing only PDFs to be uploaded so test that too
- Do other stuff that you think might break it and report if anything is weird
